File: packages/deep-engine/src/runtimePackage/dashboardTextInputTypes.ts
/** 离线单行文本筛选：字体原始字节允许运行时生成任意受该字体覆盖的字形。 */
export interface DashboardTextInputV1 {
  readonly kind: "text-v1";
  readonly nodeId: string;
  readonly key: string;
  readonly locale: "zh-CN" | "en-US";
  readonly match: "contains" | "exact";
  readonly maxGraphemes: 256;
  readonly fonts: readonly { readonly sha256: string; readonly faceIndex: number; readonly dataBase64: string }[];
  readonly style: { readonly fontSize: number; readonly lineHeight: number; readonly fontWeight: number;
    readonly fontStyle: "normal" | "italic" | "oblique"; readonly color: readonly [number, number, number, number] };
  readonly bindings: readonly { readonly nodeId: string; readonly datasetId: string;
    readonly rows: readonly (readonly (string | number | boolean | null)[])[] }[];
}

File: packages/deep-engine-native/src/runtime_package/dashboard_text_input_types.rs
use serde::{Deserialize, Serialize};
#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(deny_unknown_fields, rename_all = "camelCase")]
pub struct DashboardTextInput {
    pub kind: String,
    pub node_id: String,
    pub key: String,
    pub locale: String,
    pub r#match: String,
    pub max_graphemes: usize,
    pub fonts: Vec<InputFont>,
    pub style: InputStyle,
    pub bindings: Vec<InputBinding>,
}
#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(deny_unknown_fields, rename_all = "camelCase")]
pub struct InputFont {
    pub sha256: String,
    pub face_index: u32,
    pub data_base64: String,
}
#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(deny_unknown_fields, rename_all = "camelCase")]
pub struct InputStyle {
    pub font_size: f32,
    pub line_height: f32,
    pub font_weight: u16,
    pub font_style: crate::platform_text::TextFontStyle,
    pub color: [u8; 4],
}
#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(deny_unknown_fields, rename_all = "camelCase")]
pub struct InputBinding {
    pub node_id: String,
    pub dataset_id: String,
    pub rows: crate::chart::ChartRows,
}
